app/src/ui-blokada/kotlin/core/Dns.kt

```diff
@@ -191,23 +202,36 @@ class DnsChoicePersistence(xx: Environment) : PersistenceWithSerialiser<List<Dns
 
 }
 
-private fun addressToIpString(it: InetSocketAddress) =
-        it.hostString + ( if (it.port != 53) ":" + it.port.toString() else "" )
+private fun addressToIpString(it: InetSocketAddress, dotEnabled: Boolean) =
```

Comment:
The encoding and decoding (`addressToIpString`, `ipStringToAddress`, and `ipStringToDotEnabled`) is pretty terrible. To support backwards compatibility, the DoT hostname is appended to the IP address, with "#" as the delimiter. So:
```
8.8.8.8#dns.google;8.8.4.4#dns.google
```
This is obviously not great, any suggestions here would be appreciated.

Reply:
...and I just realized that, if the DNS list at [here](https://blokada.org/api/v4/content/en/dns.txt) is updated to include the DoT hostnames, it will break for anyone using an earlier version of Blokada (since it won't expect the hostname to be appended to the IP address). The best solution would probably be to create a new dns.txt with a separate field for the TLS hostname, and use that.

Reply:
I'm currently rewriting the API to use JSON files instead of the current format so this won't be a problem once I'm finished and the next API version is used.

Reply:
As soon as this got approval you can adapt your PR. :)  https://github.com/blokadaorg/blokada/pull/575